dartagnan/src/test/java/com/dat3m/dartagnan/comparison/AbstractComparisonTest.java

```diff
@@ -0,0 +1,133 @@
+import com.dat3m.dartagnan.utils.rules.Provider;
+import com.dat3m.dartagnan.utils.rules.Providers;
+import com.dat3m.dartagnan.utils.rules.RequestShutdownOnError;
+import com.dat3m.dartagnan.verification.VerificationTask;
+import com.dat3m.dartagnan.verification.solving.AssumeSolver;
+import com.dat3m.dartagnan.wmm.Wmm;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.rules.Timeout;
+import org.sosy_lab.common.ShutdownManager;
+import org.sosy_lab.common.configuration.Configuration;
+import org.sosy_lab.java_smt.SolverContextFactory.Solvers;
+import org.sosy_lab.java_smt.api.ProverEnvironment;
+import org.sosy_lab.java_smt.api.SolverContext;
+import org.sosy_lab.java_smt.api.SolverContext.ProverOptions;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static com.dat3m.dartagnan.configuration.OptionNames.INITIALIZE_REGISTERS;
+import static com.dat3m.dartagnan.configuration.OptionNames.USE_INTEGERS;
+import static com.dat3m.dartagnan.utils.ResourceHelper.getRootPath;
+import static java.util.Collections.emptyList;
+import static org.junit.Assert.assertEquals;
+
+public abstract class AbstractComparisonTest {
+
+    private String path;
+
+    AbstractComparisonTest(String path) {
+        this.path = path;
+    }
+
+    static Iterable<Object[]> buildLitmusTests(String litmusPath) throws IOException {
+        Set<String> skip = ResourceHelper.getSkipSet();
+        try (Stream<Path> fileStream = Files.walk(Paths.get(getRootPath(litmusPath)))) {
+            return fileStream
+                    .filter(Files::isRegularFile)
+                    .map(Path::toString)
+                    .filter(f -> f.endsWith("litmus"))
+                    .filter(f -> !skip.contains(f))
+                    .collect(ArrayList::new,
+                            (l, f) -> l.add(new Object[]{f}), ArrayList::addAll);
+        }
+    }
+
+
+    // =================== Modifiable behavior ====================
+
+    protected abstract Provider<Arch> getSourceProvider();
+    protected Provider<Wmm> getSourceWmmProvider() {
+        return Providers.createWmmFromArch(getSourceProvider());
+    }
+    protected abstract Provider<Arch> getTargetProvider();
+    protected Provider<Wmm> getTargetWmmProvider() {
+        return Providers.createWmmFromArch(getTargetProvider());
+    }
+    protected Provider<EnumSet<Property>> getPropertyProvider() {
+        return Provider.fromSupplier(() -> EnumSet.of(Property.PROGRAM_SPEC));
+    }
+    protected long getTimeout() { return 10000; }
+    // List of tests that are known to show bugs in the compilation scheme and thus the expected result should be FAIL instead of PASS
+    protected List<String> getCompilationBreakers() { return emptyList(); }
```

Comment:
This is probably a copy/paste error from the test that compares a C-like program and its compiled version. This can be probably be removed.

Reply:
Removed